frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts

```diff
@@ -223,6 +247,62 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
                 setPause: () => SessionPlayerState.PAUSE,
             },
         ],
+        playingTimeTracking: [
+            { isPlaying: false as boolean, lastTimestamp: null as number | null, watchTime: 0 },
+            {
+                setPlay: (state) => {
+                    return {
+                        isPlaying: true,
+                        lastTimestamp: state.lastTimestamp || performance.now(),
+                        watchTime: state.watchTime,
+                    }
+                },
+                setPause: (state) => {
+                    return {
+                        isPlaying: false,
+                        lastTimestamp: null,
+                        watchTime:
+                            state.lastTimestamp !== null
+                                ? state.watchTime + (performance.now() - state.lastTimestamp)
+                                : state.watchTime,
+                    }
+                },
+                setEndReached: (state, { reached }) => {
+                    if (!reached) {
+                        return state
+                    }
+
+                    return {
+                        isPlaying: false,
+                        lastTimestamp: null,
+                        watchTime:
+                            state.lastTimestamp !== null
+                                ? state.watchTime + (performance.now() - state.lastTimestamp)
+                                : state.watchTime,
+                    }
+                },
+                setErrorPlayerState: (state, { show }) => {
+                    if (!show) {
+                        return state
+                    }
+                    return {
+                        isPlaying: state.isPlaying,
+                        lastTimestamp: null,
+                        watchTime:
+                            state.lastTimestamp !== null
+                                ? state.watchTime + (performance.now() - state.lastTimestamp)
+                                : state.watchTime,
+                    }
+                },
+                seekToTime: (state) => {
+                    return {
+                        ...state,
+                        lastTimestamp:
+                            state.isPlaying && state.lastTimestamp === null ? performance.now() : state.lastTimestamp,
```

Comment:
If the replay is playing and there is a `lastTimestamp` set we are keeping it. In that case does it mean that if someone skips to a point in the video and pauses from there we'll count that whole portion towards the `watchTime`?

Reply:
It should be that when they pause we'd then set lastTimestamp to null https://github.com/PostHog/posthog/pull/16829/files/15a158b562708770c729e18417c7fa445fc38274#diff-206beaa9ee2cad5becf684afbe170be6f8fc04568a6ac02b74ed30fd7b7a2712R263

So, we'd count the time between clicking on the seek bar and the pause as part of the watch time

Reply:
Ah ok, didn't realise we called pause during the seek